internal/collector/event.go

```diff
@@ -169,3 +172,9 @@ func filterInvolvedObjectNs(list *metav1.ListOptions, ns string) {
 		list.FieldSelector += ",involvedObject.namespace=" + ns
 	}
 }
+
+func filterEventType(list *metav1.ListOptions, eventType string) {
+	if eventType != options.EventTypesAll {
+		list.FieldSelector += ",type=" + eventType
```

Comment:
Is there any way possible for user to inject anything here malicious?

Reply:
That's a very good question, thank you for pointing that out. AFAIK, the selectors are separated from one another by boolean AND which means that any malicious injection would end up hardening the filter.
That being said, I don't think it is a problem.